instruments/cryomagnetics/cryomagnetics_lm510.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+"""
+Contains an implementation of the cryomagnetics LM 510 liquid cryogen level
+monitor for InstrumentKit
+"""
+
+# IMPORTS #####################################################################
+from __future__ import absolute_import
+
+import re
+from threading import Lock
+from time import sleep
+
+import quantities as pq
+
+from instruments.abstract_instruments import CryomagneticsInstrument
+
+# CLASSES #####################################################################
+
+
+class CryomagneticsLM510(CryomagneticsInstrument):
+    """
+    Represents a Cryomagnetics LM-510 liquid cryogen level monitor
+    """
+    CHANNELS = {1, 2}
+
+    channel_measurement_lock = Lock()  # type: Lock
+    querying_lock = Lock()  # type: Lock
+
+    measurement_timeout = 1
+
+    UNITS = {
+        "cm": pq.cm,
+        "in": pq.inch,
+        "%": pq.percent,
+        "percent": pq.percent
+    }
+
+    @property
+    def default_channel(self):
+        """
+        The Level meter has a default channel that is used to measure
+        different cryogen levels. If a channel is not specified for a
+        particular command, the command gets sent to the value of the
+        default channel instead
+
+        :return: The current default channel
+        :rtype: int
+        """
+        return int(self.query("CHAN?"))
+
+    @default_channel.setter
+    def default_channel(self, channel):
+        """
+        Set the default channel to the desired channel
+
+        :param int channel: The desired channel
+        :raises: :exc:`ValueError` if the channel to be set is not in the
+        set of allowed channels defined in ``CHANNELS``
+        """
+        if channel not in self.CHANNELS:
+            raise ValueError("Attempted to set channel to %s. Channel must "
+                             "be an integer of either 1 or 2", channel)
+        command = "CHAN %d" % channel
+        self.query(command)
+
+    @property
+    def status_byte(self):
+        """
+        Get a byte array representing the instrument's status byte. The
+        status byte describes whether channel 1 or 2 are ready to make a
+        measurement. The structure of the status byte is described in
+        greater detail in the level meter's operating manual.
+
+        :return: The current value of the status byte
+        :rtype: bytes
+        """
+        byte_as_string = self.query("*STB?")
+        return byte_as_string.encode('ascii')
+
+    @property
+    def channel_1_data_ready(self):
```

Comment:
I'll have to see if the units are independently adjustable for each measurement channel. If so, there will be lots more that will have to be moved into the channel.